core/frontend/src/render/webgl/RealityMesh.ts

```diff
@@ -233,36 +233,37 @@ export class RealityMeshGeometry extends IndexedGeometry implements IDisposable,
   public get overrideColorMix() { return .5; }     // This could be a setting from either the mesh or the override if required.
   public get transform(): Transform | undefined { return this._transform; }
 
-  private constructor(private _realityMeshParams: RealityMeshGeometryParams, public textureParams: RealityTextureParams | undefined, private readonly _transform: Transform | undefined, public readonly baseColor: ColorDef | undefined, private _baseIsTransparent: boolean, private _isTerrain: boolean) {
+  private constructor(private _realityMeshParams: RealityMeshGeometryParams, public textureParams: RealityTextureParams | undefined, private readonly _transform: Transform | undefined, public readonly baseColor: ColorDef | undefined, private _baseIsTransparent: boolean, private _isTerrain: boolean, private _disableTextureDisposal?: true) {
     super(_realityMeshParams);
     this.hasTextures = undefined !== textureParams && textureParams.params.some((x) => undefined !== x.texture);
   }
 
   public override dispose() {
     super.dispose();
     dispose(this._realityMeshParams);
-    dispose(this.textureParams);
+    if (true !== this._disableTextureDisposal)
+      dispose(this.textureParams);
   }
 
-  public static createFromTerrainMesh(terrainMesh: TerrainMeshPrimitive, transform: Transform | undefined) {
+  public static createFromTerrainMesh(terrainMesh: TerrainMeshPrimitive, transform: Transform | undefined, disableTextureDisposal?: true) {
     const params = RealityMeshGeometryParams.createFromRealityMesh(terrainMesh);
-    return params ? new RealityMeshGeometry(params, undefined, transform, undefined, false, true) : undefined;
+    return params ? new RealityMeshGeometry(params, undefined, transform, undefined, false, true, disableTextureDisposal) : undefined;
   }
 
-  public static createFromRealityMesh(realityMesh: RealityMeshPrimitive): RealityMeshGeometry | undefined {
+  public static createFromRealityMesh(realityMesh: RealityMeshPrimitive, disableTextureDisposal?: true): RealityMeshGeometry | undefined {
     const params = RealityMeshGeometryParams.createFromRealityMesh(realityMesh);
     if (!params)
       return undefined;
     const texture = realityMesh.texture ? new TerrainTexture(realityMesh.texture, realityMesh.featureID, Vector2d.create(1.0, -1.0), Vector2d.create(0.0, 1.0), Range2d.createXYXY(0, 0, 1, 1), 0, 0) : undefined;
 
-    return new RealityMeshGeometry(params, texture ? RealityTextureParams.create([texture]) : undefined, undefined, undefined, false, false);
+    return new RealityMeshGeometry(params, texture ? RealityTextureParams.create([texture]) : undefined, undefined, undefined, false, false, disableTextureDisposal);
```

Comment:
These function calls were already unintelligible with six arguments, adding a seventh makes it worse.

Reply:
I will clean it up to use a Props interface for the internal constructor.

Reply:
Cleaned the constructor up to use a props object.